src/__tests__/deprecation.test.ts

```diff
@@ -1,30 +1,46 @@
-import { IsString, applyTransformer, withDeprecationWarnings } from '..';
+import {
+    IsString as DefaultIsString,
+    applyTransformer,
+    withDeprecationWarnings,
+} from '..';
 
 describe('warnIfDeprecated', () => {
     it('produces new decorator factories', () => {
-        const onWarning = jest.fn();
+        const spy = jest.spyOn(process, 'emitWarning');
```

Comment:
When we `emitWarning`, you get something like this in the console:

```
(node:45218) DeprecationWarning: Example.foo is deprecated.
(Use `node --trace-deprecation ...` to show where the warning was created)
(node:45218) DeprecationWarning: Example.baz is deprecated.
```

We can (read: should) subscribe our logger to `warning` events. We also should be able to use `--trace-deprecation` to get a stack trace of where the usage is (or even `--throw-deprecation` to fail hard).